test/detail/test_type_name.cpp

```diff
@@ -7,15 +7,17 @@
 // http://www.boost.org/LICENSE_1_0.txt)
 
 #include <turtle/detail/type_name.hpp>
-#include <boost/test/auto_unit_test.hpp>
-#include <boost/lexical_cast.hpp>
+#include <boost/test/unit_test.hpp>
+#include <sstream>
 
 namespace
 {
     template< typename T >
-    std::string to_string( const T& )
+    std::string to_string( const T& t)
     {
-        return boost::lexical_cast< std::string >( MOCK_TYPE_NAME(T) );
+        std::ostringstream s;
+        s << mock::detail::make_type_name(t); // Typename can be streamed
```

Comment:
`Typename` -> `T`?

Reply:
No I actually meant "typename" the name of the type of `t` which is returned by `make_type_name`

Reply:
Right, so maybe `// name of the type of T can be streamed`?

Reply:
I just removed the comment. I mean: It is streamed so obviously it can be streamed ;D